zypper-migration

```diff
@@ -453,11 +478,15 @@ begin
 
   raise "Interrupted." if interrupted
 
+  filtered_services = get_available_services_based_on_url
+
   migration.each do |p|
     msg = "Upgrading product #{p.friendly_name}"
     print "#{msg}.\n" unless options[:quiet]
     service = SUSE::Connect::YaST.upgrade_product p
 
+    update_services_no_path(filtered_services, service)
```

Comment:
Rather than deleting items from `services` inside `update_services_no_path`, consider returning a hash from `get_available_services_based_on_url` with service name as a key.

Then simply do `filtered_services.delete(service[:name])` here instead.